src/protocol/sort/reshare.rs

```diff
@@ -213,5 +225,69 @@ mod tests {
                 assert_eq!(secret, new_shares.reconstruct());
             }
         }
+
+        async fn reshare_with_additive_attack<F: Field>(
+            ctx: MaliciousContext<'_, F>,
+            input: &MaliciousReplicated<F>,
+            record_id: RecordId,
+            to_helper: Role,
+            additive_error: F,
+        ) -> Result<MaliciousReplicated<F>, Error> {
+            use crate::protocol::context::SpecialAccessToMaliciousContext;
+            use crate::secret_sharing::ThisCodeIsAuthorizedToDowngradeFromMalicious;
+            let rx_ctx = ctx.narrow("reshare_mac");
+            let (x, rx) = try_join(
+                ctx.narrow("reshare_input").semi_honest_context().reshare(
+                    input.x().access_without_downgrade(),
+                    record_id,
+                    to_helper,
+                ),
+                rx_ctx
+                    .semi_honest_context()
+                    .reshare(input.rx(), record_id, to_helper),
+            )
+            .await?;
+            let malicious_input =
+                MaliciousReplicated::new(x + &Replicated::new(additive_error, additive_error), rx);
```

Comment:
So this is not actually the place where the additive attack happens - that's on the communication with the other helpers - where it sends a value.

Reply:
U are right. Let me change this